pkg/types/ccipocr3/interfaces.go

```diff
@@ -18,6 +20,10 @@ type MessageHasher interface {
 	Hash(context.Context, Message) (Bytes32, error)
 }
 
+type ExtraDataCodec interface {
+	DecodeExtraData(ExtraArgs Bytes, sourceChainSelector ChainSelector) (map[string]any, error)
```

Comment:
Might be useful to put the encode function in here as well and the hasher can use it too.

Reply:
As of now, we don't have a use-case for encoding the entire ExtraArgs.
The hasher encodes only specific fields from the full ExtraArgs, so the hasher actually needs to use this decoded type.

We can add encode later if needed. I suspect before we do it, we may refactor the overall Codec interface and change these anyways.